test/Npgsql.Tests/Types/CompositeTests.cs

```diff
@@ -411,11 +416,31 @@ public void TableAsComposite()
             }
         }
 
-        class TableAsCompositeType
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/1267")]
+        public void TableAsCompositeWithDeleteColumns()
         {
-            public int Foo { get; set; }
+            using (var conn = OpenConnection())
+            {
+                conn.ExecuteNonQuery(@"
+                    DROP TABLE IF EXISTS table_as_composite CASCADE;
+                    CREATE TABLE table_as_composite (foo int, bar int);
```

Comment:
Please use `CREATE TEMP TABLE` rather than `DROP TABLE IF EXISTS ...; CREATE TABLE`. Temporary tables are automatically dropped when the connection is closed.